### PR TITLE
fix: update internal usage of useModalAttributes to use legacyTrapFocus by default

### PR DESCRIPTION
**Pull request by smhigley**
## Context

The initial decision around whether to use a hard focus trap (i.e. forcing focus back within a modal when it leaves) vs. a soft one (preventing focus from reaching the background page, but allowing it to reach browser controls) was based around where HTML seemed to be going -- both the `dialog` element and the `inert` attribute allow you to reach the browser chrome.

However, both our v9 Dialog and Popover already use the hard focus trap by default now, and user feedback seems to indicate that's preferred anyway. This PR updates `DatePicker` to be consistent with Dialog & Popover, and updates the `useModalAttributes` hook to also implement a hard focus trap by default. That seems like the best approach for both internal consistency in v9, and for keyboard & screen reader users.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `211.741 kB`<br />`60.957 kB` | `211.744 kB`<br />`60.961 kB` | `3 B` <br />`4 B` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.094 MB`<br />`270.62 kB` | `1.094 MB`<br />`270.623 kB` | `3 B` <br />`3 B` |
| <samp>react-datepicker-compat</samp> <br /> <abbr title='bundle-size/DatePicker.fixture.js'>DatePicker Compat</abbr> | `223.007 kB`<br />`63.137 kB` | `223.01 kB`<br />`63.138 kB` | `3 B` <br />`1 B` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr> | `98.898 kB`<br />`29.801 kB` | `98.901 kB`<br />`29.805 kB` | `3 B` <br />`4 B` |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr> | `127.198 kB`<br />`39.776 kB` | `127.201 kB`<br />`39.779 kB` | `3 B` <br />`3 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr> | `99.097 kB`<br />`30.197 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr> | `49.301 kB`<br />`15.838 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr> | `20.107 kB`<br />`7.973 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr> | `63.445 kB`<br />`20.061 kB` |
| <samp>react-breadcrumb</samp> <br /> <abbr title='bundle-size/index.fixture.js'>@fluentui/react-breadcrumb - package</abbr> | `113.657 kB`<br />`31.532 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr> | `37.105 kB`<br />`10.785 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr> | `43.516 kB`<br />`12.076 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr> | `41.886 kB`<br />`12.133 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr> | `49.898 kB`<br />`13.725 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr> | `53.035 kB`<br />`12.546 kB` |
| <samp>react-calendar-compat</samp> <br /> <abbr title='bundle-size/Calendar.fixture.js'>Calendar Compat</abbr> | `149.583 kB`<br />`39.839 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr> | `100.845 kB`<br />`28.873 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr> | `93.629 kB`<br />`27.077 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr> | `14.356 kB`<br />`5.798 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr> | `16.879 kB`<br />`6.664 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr> | `14.42 kB`<br />`5.934 kB` |
| <samp>react-checkbox</samp> <br /> <abbr title='bundle-size/Checkbox.fixture.js'>Checkbox</abbr> | `35.105 kB`<br />`12.112 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr> | `101.39 kB`<br />`33.324 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr> | `102.028 kB`<br />`33.263 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `69.141 kB`<br />`20.157 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.442 kB`<br />`14.607 kB` |
| <samp>react-field</samp> <br /> <abbr title='bundle-size/Field.fixture.js'>Field</abbr> | `23.384 kB`<br />`8.903 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr> | `27.987 kB`<br />`9.446 kB` |
| <samp>react-link</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Link</abbr> | `17.193 kB`<br />`6.993 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr> | `151.246 kB`<br />`45.708 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu  (including selectable components)</abbr> | `153.932 kB`<br />`46.201 kB` |
| <samp>react-message-bar</samp> <br /> <abbr title='bundle-size/Default.fixture.js'>MessageBar (all components)</abbr> | `24.596 kB`<br />`9.18 kB` |
| <samp>react-overflow</samp> <br /> <abbr title='bundle-size/hooks.fixture.js'>hooks only</abbr> | `12.821 kB`<br />`4.813 kB` |
| <samp>react-persona</samp> <br /> <abbr title='bundle-size/Persona.fixture.js'>Persona</abbr> | `56.192 kB`<br />`17.728 kB` |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr> | `14.563 kB`<br />`5.118 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
| <samp>react-progress</samp> <br /> <abbr title='bundle-size/ProgressBar.fixture.js'>ProgressBar</abbr> | `17.086 kB`<br />`6.906 kB` |
| <samp>react-provider</samp> <br /> <abbr title='bundle-size/FluentProvider.fixture.js'>FluentProvider</abbr> | `24.618 kB`<br />`8.906 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr> | `32.673 kB`<br />`10.351 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr> | `15.76 kB`<br />`6.433 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr> | `27.723 kB`<br />`10.131 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr> | `37.17 kB`<br />`12.547 kB` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr> | `36.049 kB`<br />`11.835 kB` |
| <samp>react-swatch-picker</samp> <br /> <abbr title='bundle-size/index.fixture.js'>@fluentui/react-swatch-picker - package</abbr> | `103.495 kB`<br />`30.046 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr> | `35.301 kB`<br />`11.348 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr> | `160.297 kB`<br />`45.656 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr> | `42.482 kB`<br />`13.868 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr> | `131.102 kB`<br />`36.346 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr> | `70.336 kB`<br />`20.028 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr> | `68.979 kB`<br />`19.633 kB` |
| <samp>react-tag-picker</samp> <br /> <abbr title='bundle-size/Package.fixture.js'>@fluentui/react-tag-picker - package</abbr> | `181.136 kB`<br />`54.463 kB` |
| <samp>react-tags</samp> <br /> <abbr title='bundle-size/InteractionTag.fixture.js'>InteractionTag</abbr> | `15.167 kB`<br />`6.135 kB` |
| <samp>react-tags</samp> <br /> <abbr title='bundle-size/tag.fixture.js'>Tag</abbr> | `28.996 kB`<br />`9.52 kB` |
| <samp>react-tags</samp> <br /> <abbr title='bundle-size/TagGroup.fixture.js'>TagGroup</abbr> | `82.084 kB`<br />`24.34 kB` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr> | `26.521 kB`<br />`9.741 kB` |
| <samp>react-timepicker-compat</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr> | `104.384 kB`<br />`34.782 kB` |
| <samp>react-toast</samp> <br /> <abbr title='bundle-size/Toast.fixture.js'>Toast (including Toaster)</abbr> | `97.515 kB`<br />`29.332 kB` |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr> | `54.558 kB`<br />`19.23 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/9f13dee5c8a1e6f2c9ce99494a26ea195e8a4ce6'>9f13dee5c8a1e6f2c9ce99494a26ea195e8a4ce6</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-datepicker-compat-bc4d1aaa-945e-4107-9a30-34bbb9550319.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-datepicker-compat-bc4d1aaa-945e-4107-9a30-34bbb9550319.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31801&commitId=d194455bd9219ae356d669cf9b6864975f8b5583&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask - RTL.normal.chromium.png|0| Removed|
|Avatar Converged.badgeMask.normal.chromium.png|0| Removed|
|Avatar Converged.basic.normal.chromium.png|0| Removed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-tabster/src/hooks/useModalAttributes.ts [hunk 2]
```diff
@@ -41,7 +39,7 @@ export interface UseModalAttributesOptions {
 export const useModalAttributes = (
   options: UseModalAttributesOptions = {},
 ): { modalAttributes: TabsterTypes.TabsterDOMAttribute; triggerAttributes: TabsterTypes.TabsterDOMAttribute } => {
-  const { trapFocus, alwaysFocusable, legacyTrapFocus } = options;
+  const { trapFocus, alwaysFocusable, legacyTrapFocus = true } = options;
```
Comment (layershifter) [inline packages/react-components/react-tabster/src/hooks/useModalAttributes.ts:42]: Is it a breaking change in the behavior?

Comment (smhigley) [reply packages/react-components/react-tabster/src/hooks/useModalAttributes.ts:42]: I marked it as a minor version change -- my thought behind it not being breaking is that both behaviors are modal focus traps that prevent focus from reaching the background page -- it's just a change of opinion in which approach should be our default / the standard.

Comment (smhigley) [reply packages/react-components/react-tabster/src/hooks/useModalAttributes.ts:42]: LMK if you disagree -- we could alternatively just change the docs & DatePicker, and have it be an internal practice to set `legacyTrapFocus` to true

Comment (layershifter) [reply packages/react-components/react-tabster/src/hooks/useModalAttributes.ts:42]: @smhigley as it's a public API, IMO we should not change defaults in patches/minors. Let's go with this option:

> we could alternatively just change the docs & DatePicker, and have it be an internal practice to set legacyTrapFocus to true

Comment (smhigley) [reply packages/react-components/react-tabster/src/hooks/useModalAttributes.ts:42]: @layershifter done!

**Review by sopranopillow — APPROVED** [review]

**Review by layershifter — APPROVED** [review]

**Review by khmakoto — APPROVED** [review]

**Pull request merged**